modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -584,4 +601,70 @@ public static boolean isConnectedStreamingOperation(ComponentModel componentMode
     }
     return java.util.Optional.empty();
   }
+
+  /**
+   * Checks the given {@link ParameterGroupModel}s against the given resolved parameters and validates that the group
+   * exclusiveness is honored.
+   *
+   * @param model                 the model that owns the parameter groups
+   * @param groups                the parameter groups in the model whose exclusiveness must be honored
+   * @param parameters            the resolved parameters
+   * @param aliasedParameterNames the alias for each parameter if it has one
+   *
+   * @throws ConfigurationException if the group exclusiveness is not honored by the resolved parameters.
+   *
+   * @since 4.4.0
+   */
+  public static void checkParameterGroupExclusiveness(Optional<ParameterizedModel> model,
+                                                      List<ParameterGroupModel> groups,
+                                                      Map<String, ?> parameters, Map<String, String> aliasedParameterNames)
+      throws ConfigurationException {
+    Set<String> explodedParameterNames = new HashSet<>();
+    Set<String> topLevelParameterNames = parameters.entrySet().stream()
+        .flatMap(entry -> {
+          if (entry.getValue() instanceof ParameterValueResolver) {
+            try {
+              ((ParameterValueResolver) entry.getValue()).getParameters().keySet()
+                  .forEach(k -> explodedParameterNames.add(aliasedParameterNames.getOrDefault(k, k)));
+            } catch (ValueResolvingException e) {
+              throw new MuleRuntimeException(e);
+            }
+          }
+          String key = entry.getKey();
+          aliasedParameterNames.getOrDefault(key, key);
+          return Stream.of(key);
+        })
+        .collect(toSet());
+
+    Set<String> parameterNames;
+    for (ParameterGroupModel group : groups) {
+      for (ExclusiveParametersModel exclusiveModel : group.getExclusiveParametersModels()) {
+        if (group.isShowInDsl()) {
+          parameterNames = explodedParameterNames;
```

Comment:
If we do this, and there was a pojo that is not in a ShowInDsl group, we would get its fields as parameter names here.